pkg/crcontrollers/csibmnode/controller.go

```diff
@@ -133,11 +165,12 @@ func (bmc *Controller) SetupWithManager(m ctrl.Manager) error {
 				}
 
 				k8sNode, ok := e.Object.(*coreV1.Node)
-				if !ok {
+				if !ok || !bmc.isMatchSelector(k8sNode) {
 					return false
 				}
 
-				return bmc.isMatchSelector(k8sNode)
+				bmc.enableForNode(k8sNode.Name)
```

Comment:
There is no way to remove node from such map, but I think it is not a problem since only `string` and `bool` vars will be stored in it.

Reply:
Maybe when CR is removed? It should be done during node removal procedure...

Reply:
We have to keep node name in map. If we remove it it will be added again in Update event for node and CR will be recreated again.
Let's keep potentially old string in map it won't affect performance.